### ISSUE TITLE
Add onErrorRestart operator to Eval

### ISSUE THREAD

**Issue by johnmcclean** [aol/cyclops#1005]
When an Error occurs the eval chain is restarted until it completes successfully or the specified number of retries is run down.

```java
Eval<T> onErrorRestart(long retries)
```

```java
Eval.now(10)
    .map(i -> {
                count++;
                if(count<1000)
                     throw new RuntimeException();
                return count;
        })
    .onErrorRestart(1000000)
    .get(); //1000
```

**Comment by johnmcclean** [comment]
Part of #986

**Comment by johnmcclean** [comment]
Released in 10.1.1
